### PR TITLE
Rework the bootstrap process into orchestration

### PR DESCRIPTION
**Pull request by NicolasT**
The intent of this changeset is to turn the `bootstrap.sh` script into something that invokes as little `salt` calls as possible, i.e. 'orchestrate' the whole bootstrap from Salt.

This happens in two stages:

- Bring the node into `bootstrap` state, locally. This implies running the registry, repository, `salt-master`,...
- Once `salt-master` is available, bring up the bootstrap node to a single-node control-plane through orchestration, which is *mostly*

  * Sync the master
  * Bring the bootstrap minion to a fabricated highstate
  * Deploy cluster services

### CODE REVIEW

**Comment by NicolasT** [comment]
@bert-e wait

File: salt/metalk8s/orchestrate/bootstrap/init.sls [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+Sync auth extmods:
+  salt.runner:
+  - name: metalk8s_saltutil.sync_auth
+  - saltenv: {{ saltenv }}
+
+Sync bootstrap minion:
+  salt.function:
+  - name: saltutil.sync_all
+  - tgt: {{ pillar.bootstrap_id }}
```
Comment (TeddyAndrieux) [inline salt/metalk8s/orchestrate/bootstrap/init.sls:63]: You need to specify `saltenv` here

File: salt/metalk8s/orchestrate/bootstrap/init.sls [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  - pillar: {{ pillar_data | tojson }}
+  - require:
+    - http: Wait for API server to be available
+
+Update pillar on bootstrap minion after highstate:
+  salt.function:
+  - name: saltutil.refresh_pillar
+  - tgt: {{ pillar.bootstrap_id }}
+  - require:
+    - salt: Configure bootstrap Node object
```
Comment (TeddyAndrieux) [inline salt/metalk8s/orchestrate/bootstrap/init.sls:108]: This require is a left over I think ?

**Review by TeddyAndrieux — COMMENTED** [review]
Approach LGTM

File: salt/metalk8s/orchestrate/bootstrap/init.sls (hunk 2 above)
Comment (NicolasT) [reply salt/metalk8s/orchestrate/bootstrap/init.sls:108]: No. The Pillar `top` is derived from the node/minion roles, and these are derived from annotations on the `Node` object, which needs to be configured correctly before we can continue.

File: salt/metalk8s/orchestrate/bootstrap/init.sls (hunk 1 above)
Comment (NicolasT) [reply salt/metalk8s/orchestrate/bootstrap/init.sls:63]: You're right, done.

File: salt/metalk8s/orchestrate/bootstrap/init.sls (hunk 2 above)
Comment (TeddyAndrieux) [reply salt/metalk8s/orchestrate/bootstrap/init.sls:108]: Yes indeed just when I commented the step "salt.runner Configure bootstrap Node object" just above was not here :stuck_out_tongue:

**Comment by NicolasT** [comment]
@bert-e reset

**Comment by NicolasT** [comment]
@bert-e approve

**Comment by NicolasT** [comment]
@bert-e reset

**Review by TeddyAndrieux — APPROVED** [review]
LGTM

**Comment by NicolasT** [comment]
@bert-e status

**Pull request merged**